### PR TITLE
Feat/77/send verification email

### LINKED ISSUE #77: Send Verification Email After Registration
Send an email asking them to verify their account. The email should have a link to that will verify their email once opened.

### PR DESCRIPTION
**Pull request by juancwu**
- Add a new email service function to send verification email
- Add new jwt functions to generate/verify email tokens (the token that is attached to the verification email)
- Add a new table to store any generated jwt for emails, this can be used to invalidate old tokens in old emails when new ones are generated.
- Add a simple view for the verification email

### CODE REVIEW

File: backend/internal/jwt/generate.go [hunk 1]
```diff
@@ -28,6 +32,45 @@ func Generate(userID string, role db.UserRole) (string, string, error) {
...
+	}()
+
+	select {
+	case <-ctx.Done():
+		return "", fmt.Errorf("context canceled while generating verify email JWT: %w", ctx.Err())
+	case result := <-done:
+		return result.token, result.err
+	}
+}
+
```
Comment (AmirAgassi) [inline backend/internal/jwt/generate.go]: Using a goroutine here is probably unnecessarily complex, JWT signing is pretty fast. We could probably just sign synchronously and return the result lol

File: backend/internal/jwt/verify.go [hunk 2]
```diff
@@ -22,3 +22,20 @@ func VerifyToken(token string) (*JWTClaims, error) {
...
+		if _, ok := t.Method.(*golangJWT.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", t.Header["alg"])
+		}
+		return []byte(os.Getenv("JWT_SECRET_VERIFY_EMAIL")), nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &claims, nil
+}
```
Comment (AmirAgassi) [inline backend/internal/jwt/verify.go]: Could add stricter validation for the token type (claims.TokenType == VERIFY_EMAIL_TOKEN_TYPE) to make sure it doesn't accidentally validate other JWT types

File: backend/internal/views/verification_email.templ [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+templ VerificationEmail(url string) {
+	<div>
+		<h1>SPUR x Konfer</h1>
+		<div>
+			<h1>Let's Verify Your Email</h1>
+			<p>Please click the link below and we will verify your email.</p>
+			<a href={ templ.URL(url) }>Verify now</a>
+		</div>
+	</div>
```
Comment (AmirAgassi) [inline backend/internal/views/verification_email.templ:11]: Don't know if it's still relevant, but maybe add a fallback link ("If the link doesn't work, copy paste this URL to your browser.")
Just bc I see a lot of sites do this lol

File: backend/internal/jwt/generate.go (hunk 1 above)
Comment (juancwu) [reply backend/internal/jwt/generate.go]: True, reading the code again, there is no need for the function to use a goroutine. I was trying to take use of the timeout context but actually won't really matter.

File: backend/internal/views/verification_email.templ (hunk 3 above)
Comment (juancwu) [reply backend/internal/views/verification_email.templ:11]: I want to make a separate issue to update the template and make it better both UX and UI. Right now, just get the functionality in for the prototype.

Comment (AmirAgassi) [reply backend/internal/views/verification_email.templ:11]: Fire

**Review by AmirAgassi — APPROVED** [review]
lgtm

**Pull request merged**